From 767311fbce0d59a9ad3fa49573eba649c1c60dd7 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@st.com>
Date: Fri, 20 Jan 2023 08:11:37 +0100
Subject: [PATCH] arm64: dts: st: enable Ethernetx on stm32mp257f-ev1 board

One Ethernet instance (ETHERNET2) is connected to Realtek PHY in RGMII mode
Other Ethernet instance (ETHERNET1) is connected to TTTech switch.
TTTech switch are 2 Ports, each ports are connected to Realtek PHY in
RGMII mode (eth1 and eth3).
Ethernets are SNSP IP with GMAC5 version.

Signed-off-by: Christophe Roullier <christophe.roullier@st.com>
Change-Id: I524b9c7936644cd8f6526c05f9a58b84e56c8807
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 71 ++++++++++++++++++++++
 1 file changed, 71 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
index 6c5b823b0fdc..d555add34dfa 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -19,6 +19,8 @@ / {
 	compatible = "st,stm32mp257f-ev1", "st,stm32mp257";
 
 	aliases {
+		ethernet0 = &eth2;
+		ethernet1 = &eth1;
 		serial0 = &usart2;
 	};
 
@@ -37,6 +39,67 @@ &arm_wdt {
 	status = "okay";
 };
 
+&eth1 {
+	status = "disabled";
+	pinctrl-0 = <&eth1_mdio_pins_a>;
+	pinctrl-names = "default";
+	phy-mode = "rgmii";
+	max-speed = <1000>;
+	st,eth-clk-sel;
+
+	fixed_link: fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+
+	mdio1 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "snps,dwmac-mdio";
+
+		phy1_eth1: ethernet-phy@4 {
+			compatible = "ethernet-phy-id001c.c916",
+				     "ethernet-phy-ieee802.3-c22";
+			realtek,eee-disable;
+			reg = <4>;
+		};
+
+		phy2_eth1: ethernet-phy@5 {
+			compatible = "ethernet-phy-id001c.c916",
+				     "ethernet-phy-ieee802.3-c22";
+			reset-gpios =  <&gpioj 9 GPIO_ACTIVE_LOW>;
+			reset-assert-us = <10000>;
+			reset-deassert-us = <300>;
+			realtek,eee-disable;
+			reg = <5>;
+		};
+	};
+};
+
+&eth2 {
+	status = "okay";
+	pinctrl-0 = <&eth2_rgmii_pins_a>;
+	pinctrl-names = "default";
+	phy-mode = "rgmii-id";
+	max-speed = <1000>;
+	phy-handle = <&phy1_eth2>;
+	st,eth-ptp-from-rcc;
+
+	mdio1 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "snps,dwmac-mdio";
+		phy1_eth2: ethernet-phy@1 {
+			compatible = "ethernet-phy-id001c.c916";
+			reset-gpios =  <&gpiog 6 GPIO_ACTIVE_LOW>;
+			reset-assert-us = <10000>;
+			reset-deassert-us = <300>;
+			realtek,eee-disable;
+			reg = <1>;
+		};
+	};
+};
+
 &hpdma {
 	memory-region = <&hpdma1_lli>;
 };
@@ -99,6 +162,14 @@ &sdmmc1 {
 	status = "okay";
 };
 
+&switch0 {
+	status = "disabled";
+	pinctrl-0 = <&eth1_rgmii_pins_a>, <&eth3_rgmii_pins_a>;
+	pinctrl-names = "default";
+	phy-mode = "rgmii";
+	st,ethsw-internal-125;
+};
+
 &usart2 {
 	pinctrl-names = "default", "idle", "sleep";
 	pinctrl-0 = <&usart2_pins_a>;
-- 
2.39.5

